statsd/src/test/java/com/avast/metrics/statsd/StatsDMetricsMonitorTest.java

```diff
@@ -108,4 +108,13 @@ public void failsForGaugeIfNotReplacing() {
             }
         }
     }
+
+    @Test
+    public void nameSanitizationWorks() {
+        try(final StatsDMetricsMonitor monitor = new StatsDMetricsMonitor("",1234,"com.avast.domain")) {
+            assertEquals(monitor.named("hello*").getName(), "hello_");
+            assertEquals(monitor.named("2001:***@1.00").getName(), "2001_____1.00");
```

Comment:
Why this line passes? `name.replaceAll("[^a-zA-Z0-9-]", "_");` should replace also the dot. Please add a test with big enough subset of ascii table.

Reply:
It doesn't:) Do you think that test for `{}()[]#&@;+-/*,\|` is enough?

Reply:
There are 4 ranges that won't be replaced, the loop and conditions should be quite simple...

Reply:
done